### PR TITLE
NODE-2632 Delayed Light Node block structure

### PR DESCRIPTION
**Pull request by xrtm000**

### CODE REVIEW

File: node/src/test/scala/com/wavesplatform/mining/LightNodeBlockFieldsTest.scala [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+class LightNodeBlockFieldsTest extends PropSpec with WithDomain {
+  property("new block fields appear 1000 blocks after LightNode activation") {
+    withDomain(
+      TransactionStateSnapshot.setFeaturesHeight(LightNode -> 2),
+      AddrWithBalance.enoughBalances(defaultSigner, secondSigner)
+    ) { d =>
+      withMiner(
+        d.blockchain,
+        d.testTime,
+        d.settings.copy(minerSettings = d.settings.minerSettings.copy(quorum = 0, minMicroBlockAge = 0.seconds)),
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/mining/LightNodeBlockFieldsTest.scala]:
```suggestion
  property("new block fields appear lightNodeBlockFieldsAbsenceInterval blocks after LightNode activation") {
    val settings = {
      val s = TransactionStateSnapshot.setFeaturesHeight(LightNode -> 2)
      s.copy(
        minerSettings = s.minerSettings.copy(quorum = 0, minMicroBlockAge = 0.seconds),
        blockchainSettings = s.blockchainSettings.copy(
          functionalitySettings = s.blockchainSettings.functionalitySettings.copy(
            lightNodeBlockFieldsAbsenceInterval = 10
          )
        )
      )
    }

    withDomain(
      settings,
      AddrWithBalance.enoughBalances(defaultSigner, secondSigner)
    ) { d =>
      withMiner(d.blockchain, d.testTime, d.settings, verify = false, timeDrift = Int.MaxValue) { case (miner, append) =>
```

File: node/src/test/scala/com/wavesplatform/mining/LightNodeBlockFieldsTest.scala [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        hash2 should not be hash1
+
+        d.rollbackTo(1000)
+        challengeBlock()
+        block(1001).stateHash shouldBe None
+        block(1001).challengedHeader shouldBe None
+
+        challengeBlock()
+        block(1002).stateHash shouldBe defined
+        block(1002).challengedHeader shouldBe defined
```
Comment (vsuharnikov) [inline node/src/test/scala/com/wavesplatform/mining/LightNodeBlockFieldsTest.scala]:
```suggestion
        (1 to 8).foreach(_ => appendBlock())
        d.blockchain.height shouldBe 11
        block(11).stateHash shouldBe None

        appendMicro()
        block(11).stateHash shouldBe None

        appendBlock()
        d.blockchain.height shouldBe 12
        val hash1 = block(12).stateHash
        hash1 shouldBe defined

        appendMicro()
        val hash2 = block(12).stateHash
        hash2 shouldBe defined
        hash2 should not be hash1

        d.rollbackTo(10)
        challengeBlock()
        block(11).stateHash shouldBe None
        block(11).challengedHeader shouldBe None

        challengeBlock()
        block(12).stateHash shouldBe defined
        block(12).challengedHeader shouldBe defined
```
Less blocks - faster tests :)

Comment (xrtm000) [reply node/src/test/scala/com/wavesplatform/mining/LightNodeBlockFieldsTest.scala]: Done

File: node/src/test/scala/com/wavesplatform/mining/LightNodeBlockFieldsTest.scala (hunk 1 above)
Comment (xrtm000) [reply node/src/test/scala/com/wavesplatform/mining/LightNodeBlockFieldsTest.scala]: Done

**Review by DrBlast — APPROVED** [review]

**Review by vsuharnikov — APPROVED** [review]

**Review by ivan-mashonskiy — APPROVED** [review]

**Review by phearnot — APPROVED** [review]

**Pull request merged**